### ISSUE TITLE
"DUMP" command produces content not compatible with Redis

### ISSUE THREAD

**Issue by krizald** [apache/kvrocks#2251]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Version

OS:
 Debian 11 bullseye

KV Rock
version:unstable
git_sha1:433dcfb3

### Minimal reproduce step

I checked and compiled KV Rocks from unstable (SHA: 433dcfb39b689503b747eabaf922a8718a1d4575) and test the new "DUMP" command. My plan is to restore keys from KV Rocks back to Redis 6.2.10/7.2.4 using DUMP/RESTORE

1. In KV Rocks
```
set A "ABC"`
dump A
```
Then I get the content: "\x00\x03ABC\x0c\x00\x8a\x95H\n\x1a\xf2\x0eO"
2. In Redis 6.2.10/7.2.4
```
restore A 0 "\x00\x03ABC\x0c\x00\x8a\x95H\n\x1a\xf2\x0eO"
```
I got message: (error) ERR DUMP payload version or checksum are wrong

### What did you expect to see?

I ran the same commands in Redis 6.2.10
```
set A "ABC"
dump A
```
What I got is: 
"\x00\x03ABC\t\x00\x91\x11-\x06\xb4^\xf3\xb7"
And when I restore, it works

### What did you see instead?

For same string value "ABC", KV Rocks dumps: 
`"\x00\x03ABC\x0c\x00\x8a\x95H\n\x1a\xf2\x0eO"`
Redis 6.2.10 is 
`"\x00\x03ABC\t\x00\x91\x11-\x06\xb4^\xf3\xb7"`

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by krizald** [comment]
more over, KV Rocks is compatible with both dump content

```
127.0.0.1:7381> restore B 0 "\x00\x03ABC\x0c\x00\x8a\x95H\n\x1a\xf2\x0eO"
OK
127.0.0.1:7381> get B
"ABC"
127.0.0.1:7381> restore C 0 "\x00\x03ABC\t\x00\x91\x11-\x06\xb4^\xf3\xb7"
OK
127.0.0.1:7381> get C
"ABC
```

**Comment by git-hulk** [comment]
@krizald Thanks for your report, I will take a look while I get time.

**Comment by git-hulk** [comment]
@krizald This issue would be fixed in #2252
